src/mpid/ch4/netmod/ucx/ucx_impl.h

```diff
@@ -33,7 +33,7 @@ static inline uint64_t MPIDI_UCX_init_tag(MPIR_Context_id_t contextid, int sourc
     ucp_tag = (ucp_tag << MPIDI_UCX_SOURCE_SHIFT);
     ucp_tag |= source;
     ucp_tag = (ucp_tag << MPIDI_UCX_TAG_SHIFT);
-    ucp_tag |= (MPIDI_UCX_TAG_MASK & tag);
+    ucp_tag |= tag;
```

Comment:
Won't this overwrite some of the source bits? Can we add a Min's tag_ub test to this PR to verify everything is working correctly?

Reply:
Come to think of it, the tag_ub test won't necessarily fail here. I still think we should test it, though.

But I think the issue is clear. The tag mask now covers 31 bits after modifying above. Removing its use here means we will set the lower 32 bits in `ucp_tag`, one more than we should.